csharp/src/LibUSearch/USearchIndex.cs

```diff
@@ -0,0 +1,262 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using static NativeMethods;
+
+public class USearchIndex : IDisposable
+{
+    private IntPtr _index;
+    private bool _disposedValue = false;
+    private ulong _cached_dimensions;
+
+    public USearchIndex(
+        usearch_metric_kind_t metricKind,
```

Comment:
Does this enum type name match C# conventions? Looks a bit weird :)

Reply:
I became confused by the guidelines in the official documents. I recently made a commit in which I refactored the user-visible names in `USearchIndex` to match the camelCase style used in C#.

Reply:
@ashvardanian @AleksandrKent  just in case, I could run the C# code through our usual code style checker and send a PR if anything needed. The code style is a mix of .editorconfig rules and resharper settings. .editorconfig settings can be easily tested from CLI with `dotnet format`, while resharper checks are easier with a manual check just once, opening Rider (or you could ignore resharper)